mmdet/models/detectors/deformable_detr.py

```diff
@@ -539,3 +539,29 @@ def get_proposal_pos_embed(proposals: Tensor,
         pos = torch.stack((pos[:, :, :, 0::2].sin(), pos[:, :, :, 1::2].cos()),
                           dim=4).flatten(2)
         return pos
+
+    def _forward(
+            self,
+            batch_inputs: Tensor,
+            batch_data_samples: OptSampleList = None) -> Tuple[List[Tensor]]:
+        """Network forward process. Usually includes backbone, neck and head
+        forward without any post-processing.Overwrite to pop
+        'enc_outputs_class' and 'enc_outputs_coord'.
+
+         Args:
+            batch_inputs (Tensor): Inputs, has shape (bs, dim, H, W).
+            batch_data_samples (List[:obj:`DetDataSample`], optional): The
+                batch data samples. It usually includes information such
+                as `gt_instance` or `gt_panoptic_seg` or `gt_sem_seg`.
+                Defaults to None.
+
+        Returns:
+            tuple[Tensor]: A tuple of features from ``bbox_head`` forward.
+        """
+        img_feats = self.extract_feat(batch_inputs)
+        head_inputs_dict = self.forward_transformer(img_feats,
+                                                    batch_data_samples)
+        head_inputs_dict.pop('enc_outputs_class')
+        head_inputs_dict.pop('enc_outputs_coord')
+        results = self.bbox_head.forward(**head_inputs_dict)
+        return results
```

Comment:
You do not need to write this
There was process here in `self.pre_decoder`.
It is enough to complete this part of the logic.

```Python
def pre_decoder(self, memory: Tensor, memory_mask: Tensor,
                spatial_shapes: Tensor) -> Tuple[Dict, Dict]:
    ......
    head_inputs_dict = dict(
        enc_outputs_class=enc_outputs_class if self.as_two_stage else None,
        enc_outputs_coord=enc_outputs_coord_unact.sigmoid()
        if self.as_two_stage else None)
    return decoder_inputs_dict, head_inputs_dict
```

Reply:
maybe you should try to run it,as _forward does not have enclass and encoord